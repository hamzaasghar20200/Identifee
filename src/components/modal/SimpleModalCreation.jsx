import {
  Modal,
  ModalBody,
  ModalFooter,
  ModalHeader,
  Spinner,
} from 'reactstrap';
import { overflowing, removeBodyScroll } from '../../utils/Utils';
import ButtonIcon from '../commons/ButtonIcon';

export default function SimpleModalCreation({
  children,
  modalTitle,
  open,
  handleSubmit,
  buttonsDisabled,
  onHandleCloseModal,
  errorMessage,
  setErrorMessage,
  isLoading,
  editFields,
  saveButton,
  deleteButton,
  deleteButtonLoading,
  successMessage,
  loading,
  setSuccessMessage,
  customModal,
  bodyClassName = 'mb-0 p-3',
  bankTeam = false,
  toggle,
  noFooter = false,
  saveButtonStyle = 'btn-primary',
  ...rest
}) {
  return (
    <Modal
      onOpened={() => {
        removeBodyScroll();
      }}
      onClosed={() => {
        overflowing();
      }}
      isOpen={open}
      fade={false}
      {...rest}
      className={customModal}
    >
      <ModalHeader
        tag="h3"
        toggle={onHandleCloseModal}
        className={modalTitle ? 'p-3 text-capitalize' : 'p-0'}
      >
        {modalTitle}
      </ModalHeader>
      <ModalBody
        className={`${bodyClassName} ${modalTitle ? 'border-top' : ''}`}
      >
        {children}
      </ModalBody>

      {(bankTeam === false || noFooter === false) && (
        <ModalFooter
          className={`px-3 ${
            deleteButton ? 'justify-content-between' : 'justify-content-end'
          }`}
        >
          {deleteButton && (
            <ButtonIcon
              icon={deleteButton.icon}
              label={deleteButton.label}
              loading={deleteButtonLoading}
              onclick={deleteButton.onClick}
              color="outline-danger"
              classnames={`btn-sm pull-left ${
                deleteButton.show ? '' : 'd-none'
              }`}
            />
          )}
          <div className="d-flex gap-1 justify-content-end align-items-center">
            {onHandleCloseModal && (
              <button
                className="btn btn-white btn-sm"
                data-dismiss="modal"
                onClick={onHandleCloseModal}
                disabled={isLoading}
              >
                Cancel
              </button>
            )}
            {handleSubmit && (
              <button
                type="button"
                className={`btn btn-sm ${saveButtonStyle}`}
                onClick={handleSubmit}
                disabled={isLoading || buttonsDisabled}
              >
                {isLoading ? (
                  <Spinner className="spinner-grow-xs" />
                ) : (
                  <span>{editFields ? 'Update' : saveButton || 'Save'}</span>
                )}
              </button>
            )}
          </div>
        </ModalFooter>
      )}
    </Modal>
  );
}
