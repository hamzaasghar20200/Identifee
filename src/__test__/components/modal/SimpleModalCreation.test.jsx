import { render, screen } from '@testing-library/react';

import SimpleModalCreation from '../../../components/modal/SimpleModalCreation';

describe('SimpleModalCreation', () => {
  const open = true;
  const modalTitle = 'Modal Test';
  const onHandleCloseModal = jest.fn();
  const handleSubmit = jest.fn();

  const children = <div>From test</div>;

  test('snapshot', () => {
    const { baseElement } = render(
      <SimpleModalCreation open={open} modalTitle={modalTitle}>
        {children}
      </SimpleModalCreation>
    );

    expect(baseElement).toMatchSnapshot();
  });

  test('Should we can to see a modal', () => {
    const { baseElement } = render(
      <SimpleModalCreation open={open} modalTitle={modalTitle}>
        {children}
      </SimpleModalCreation>
    );

    const modalClasses = baseElement.getElementsByClassName('modal-dialog');

    expect(modalClasses).toHaveLength(1);

    const modalContentClasses =
      baseElement.getElementsByClassName('modal-content');

    expect(modalContentClasses).toHaveLength(1);
  });

  test('Should we can to see a title', () => {
    const component = render(
      <SimpleModalCreation open={open} modalTitle={modalTitle}>
        {children}
      </SimpleModalCreation>
    );

    const titleTag = component.getByText(modalTitle);

    expect(titleTag).toBeTruthy();
  });

  test('Should have 1 child', () => {
    const { baseElement } = render(
      <SimpleModalCreation open={open} modalTitle={modalTitle}>
        {children}
      </SimpleModalCreation>
    );

    const child = baseElement.getElementsByClassName('modal-body')[0];

    expect(child.childElementCount).toEqual(1);
  });

  test('Should have a footer and buttons', () => {
    render(
      <SimpleModalCreation
        open={open}
        modalTitle={modalTitle}
        onHandleCloseModal={onHandleCloseModal}
        handleSubmit={handleSubmit}
      >
        {children}
      </SimpleModalCreation>
    );

    const cancelButton = screen.getByText('Cancel');

    expect(cancelButton).toBeTruthy();

    const saveButton = screen.getByText('Save');

    expect(saveButton).toBeTruthy();
  });

  test('Should show an update button', () => {
    render(
      <SimpleModalCreation
        open={open}
        modalTitle={modalTitle}
        onHandleCloseModal={onHandleCloseModal}
        handleSubmit={handleSubmit}
        editFields={true}
      >
        {children}
      </SimpleModalCreation>
    );

    const updateButton = screen.getByText('Update');

    expect(updateButton).toBeTruthy();
  });

  test('Should show a spinner when we have loading', () => {
    const { baseElement } = render(
      <SimpleModalCreation
        open={open}
        modalTitle={modalTitle}
        onHandleCloseModal={onHandleCloseModal}
        handleSubmit={handleSubmit}
        isLoading={true}
      >
        {children}
      </SimpleModalCreation>
    );

    const loading = baseElement.getElementsByClassName('spinner-border');

    expect(loading).toHaveLength(1);
  });
});
