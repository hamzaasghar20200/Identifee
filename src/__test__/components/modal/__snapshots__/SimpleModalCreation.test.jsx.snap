// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleModalCreation snapshot 1`] = `
<body
  class="modal-open"
>
  <div />
  <div
    style="position: relative; z-index: 1050; display: block;"
    tabindex="-1"
  >
    <div
      class=""
    >
      <div
        class="modal fade"
        role="dialog"
        style="display: block;"
        tabindex="-1"
      >
        <div
          class="modal-dialog"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h2
                class="modal-title"
              >
                Modal Test
              </h2>
            </div>
            <div
              class="modal-body"
            >
              <div>
                From test
              </div>
            </div>
            <div
              class="modal-footer"
            />
          </div>
        </div>
      </div>
      <div
        class="modal-backdrop fade"
      />
    </div>
  </div>
</body>
`;
